List Them Out - A Modern Revision On Extracting Global Lists

san7890 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Hey there,

About 10 months ago, I asked "why the fuck is something pretty neat like 'List Bombers' in the Secrets panel?" Isn't that kinda obtuse? Kinda goofy? Silly, even? It's just because the Secrets panel was ancient, even if it got that new coat of paint. So, let's clean it up!

I moved all of the functions (which are fancy ways of pulling entries out of a list) to their own admin procs, and I span verbs out of those procs. So, now they show up in the admin panel like this:

![image](https://user-images.githubusercontent.com/34697715/191881028-e9e8ea57-d911-4d07-867b-c22c38566352.png)

WARNING: Image outdated, only List Bombers and List Signalers show up here.

You see those in the middle there? They exist. You can also call from the command line! Very cool, right?

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

![image](https://user-images.githubusercontent.com/34697715/191881021-f5c48f6b-480f-4a2c-a4e1-00e763d864de.png)

Love those crazy lists.

I did some general code cleanup/polishing to modern code standards, as well as modernized some parts of it. I moved output_ai_laws() to the new file I created in this PR since it felt more organized given that all six of those verbs originated from the secrets panel in some way.

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
admin: List Bombers, List Law Changes, List Manifest, List Signalers, List DNA, List Fingerprints, and Show Law Changes have all been replicated to exist outside of the Secrets panel. You may still access these verbs from the Secrets panel, but they are able to be called as individual verbs.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

san7890 committed:
List Them Out - A Modern Revision On Extracting Global Lists

Hey there,

About 10 months ago, I asked "why the fuck is something pretty neat like 'List Bombers' in the Secrets panel?" Isn't that kinda obtuse? Kinda goofy? Silly, even? It's just because the Secrets panel was ancient, even if it got that new coat of paint. So, let's clean it up!

I moved all of the functions (which are fancy ways of pulling entries out of a list) to their own admin procs, and I span verbs out of those procs. So, now they show up in the admin panel like this:

As well as being able to be called from the command line! Very cool, right?

san7890 committed:
signalers, not signallers

spellcheck

san7890 committed:
fixed fucked capitalization

Mothblocks commented:
add a border=1 to these tables

san7890 committed:
adds border=1 to the tables

JohnFulpWillard commented on code/modules/admin/verbs/list_exposer.dm:
```diff
@@ -0,0 +1,89 @@
+
+/datum/admins/proc/output_ai_laws()
+	var/ai_number = 0
+	for(var/entry in GLOB.silicon_mobs)
+		var/mob/living/silicon/subject = entry
```
```suggestion
	for(var/mob/living/silicon/subject as anything in GLOB.silicon_mobs)
```

JohnFulpWillard commented on code/modules/admin/verbs/list_exposer.dm:
```diff
@@ -0,0 +1,89 @@
+	var/ai_number = 0
+	for(var/entry in GLOB.silicon_mobs)
+		var/mob/living/silicon/subject = entry
+		ai_number++
```
this implies it's AI, but cyborgs/pAIs count for it. Should it be renamed?

Iamgoofball commented on code/modules/admin/admin_verbs.dm:
```diff
@@ -79,6 +79,12 @@ GLOBAL_PROTECT(admin_verbs_admin)
 	/datum/admins/proc/paintings_manager,
 	/datum/admins/proc/display_tags,
 	/datum/admins/proc/fishing_calculator,
+	/client/proc/list_bombers,
```
why are these all separate verbs, just make one that pulls up a pick list of all of these

Watermelon914 replied:
yeah, do this
It'd be like the 'Investigate' verb, which is more compact.
Additionally you could just integrate it into the Investigate verb

san7890 replied:
I want to be able to type in `list-signallers` or `list-bombers` or even `list-law-changes` rather than have yet ANOTHER pick list (the Secrets menu already pretty much serves this function) rather than have to do some sort of advanced proc call to get to the information if secrets panel isn't working.

Mothblocks replied:
i agree with san

Iamgoofball commented on code/modules/admin/admin_verbs.dm:
```diff
@@ -349,6 +355,54 @@ GLOBAL_PROTECT(admin_verbs_poll)
+	if(!holder)
+		return
+	holder.list_bombers()
+	SSblackbox.record_feedback("tally", "admin_verb", 1, "List Bombers") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
```
IF YOU ARE COPY-PASTING THIS ENSURE THE 2ND PARAMETER IS UNIQUE TO THE NEW PROC

JohnFulpWillard replied:
admin verbs are all together, it's fine

Iamgoofball replied:
okay but ```IF YOU ARE COPY-PASTING THIS ENSURE THE 2ND PARAMETER IS UNIQUE TO THE NEW PROC```

san7890 committed:
as anything

Co-authored-by: John Willard <53777086+JohnFulpWillard@users.noreply.github.com>

san7890 committed:
updates name of that variable

law bound entities has a nice ring to it

san7890 committed:
keeps the verbs that are used in actual administrative work

the others can just go die in the debug list since they aren't really used in "Admin Needs To See What Goes On During A Game" work, but i'll leave them grouped next to each other since it's still the same section of exposing a global list into a nice window

keep in mind they will always be accessible via secrets panel though
